sdk/typescript/src/utils/ed25519-hd-key.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This is adapted from https://github.com/alepop/ed25519-hd-key replacing create-hmac 
+// with @noble/hashes to be browser compatible.
+
+import { sha512 } from '@noble/hashes/sha512';
+import { hmac } from '@noble/hashes/hmac';
+import nacl from 'tweetnacl';
+
+type Hex = string;
+type Path = string;
+
+type Keys = {
+    key: Buffer;
```

Comment:
I know this comes from the module we're forking, but `Buffer` doesn't natively exist on browsers, so we should really avoid using it.